### PR TITLE
Documentation: Fixes API comment for `RequestOptions.IfMatchEtag`

### LINKED ISSUE #3976: IfMatchEtag on ReadItemAsync not honoured
**Describe the bug**
`IfMatchEtag` does not appear to be working as described.

The documentation states:
> Most commonly used with the Delete* and Replace* methods of [Container](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.container?view=azure-dotnet) such as [ReplaceItemAsync<T>(T, String, Nullable<PartitionKey>, ItemRequestOptions, CancellationToken)](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.container.replaceitemasync?view=azure-dotnet#microsoft-azure-cosmos-container-replaceitemasync-1(-0-system-string-system-nullable((microsoft-azure-cosmos-partitionkey))-microsoft-azure-cosmos-itemrequestoptions-system-threading-cancellationtoken)) but can be used with other methods like [ReadItemAsync<T>(String, PartitionKey, ItemRequestOptions, CancellationToken)](https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.cosmos.container.readitemasync?view=azure-dotnet#microsoft-azure-cosmos-container-readitemasync-1(system-string-microsoft-azure-cosmos-partitionkey-microsoft-azure-cosmos-itemrequestoptions-system-threading-cancellationtoken)) for caching scenarios.

I have witnessed this working previously on an UpsertItem call as well as a Replace call.

However, it does not appear to work on ReadItemAsync. This can be verified relatively simply.

**To Reproduce**

```
var etag = "....";
var result = await container.ReadItemAsync<T>(id, new PartitionKey(partitionKey), new ItemRequestOptions {
    IfMatchEtag = etag
});
if (result.ETag != etag) {
    throw new Exception("Cosmos returned document with mismatched ETag");
}
```

**Expected behavior**
I would expect a `CosmosException` with either a `PreconditionFailed` or `Conflict` (although the latter is not so semantic) status.

**Actual behavior**
The item is returned without issue, with a different ETag than what was presented. Status Code is `OK`.

**Environment summary**
SDK Version: .NET 3.35.1
OS Version: Mac OSX, also verified on Linux (App Service), and Windows (another dev machine)

**Additional context**
This can be reproduced in the Javascript SDK.

### LINKED ISSUE #4250: Documentation: Fix Public API Comment in `RequestOptions` for the Proper Usage of `IfMatchEtag` Property.
- Fix Public API Comment in `RequestOptions` for the Proper Usage of `IfMatchEtag` Property.

### PR DESCRIPTION
**Pull request by kundadebdatta**
# Pull Request Template

## Description

This PR fixes the API documentation for `RequestOptions.IfMatchEtag`.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Closing issues

To automatically close an issue: closes #4250, closes #3976

### CODE REVIEW

**Review by ealsur — APPROVED** [review]
Please link issues to close them

**Review by philipthomas-MSFT — DISMISSED** [review]
LGTM

**Review by NaluTripician — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/RequestOptions/RequestOptions.cs [hunk 1]
```diff
@@ -18,8 +18,7 @@ public class RequestOptions
         /// Gets or sets the If-Match (ETag) associated with the request in the Azure Cosmos DB service.
         /// </summary>
         /// <remarks>
-        /// Most commonly used with the Delete* and Replace* methods of <see cref="Container"/> such as <see cref="Container.ReplaceItemAsync{T}(T, string, PartitionKey?, ItemRequestOptions, System.Threading.CancellationToken)"/>
-        /// but can be used with other methods like <see cref="Container.ReadItemAsync{T}(string, PartitionKey, ItemRequestOptions, System.Threading.CancellationToken)"/> for caching scenarios.
+        /// Most commonly used with the Delete* and Replace* methods of <see cref="Container"/> such as <see cref="Container.ReplaceItemAsync{T}(T, string, PartitionKey?, ItemRequestOptions, System.Threading.CancellationToken)"/>.
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/RequestOptions/RequestOptions.cs:21]: nit: How about also covering it part of ReadItem* code-doc remarks.

Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/RequestOptions/RequestOptions.cs:21]: Added the code doc on `ContainerInlineCore.ReadItemAsync()`

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerInlineCore.cs [hunk 2]
```diff
@@ -260,6 +260,8 @@ Task<ResponseMessage> func(ITrace trace)
             ItemRequestOptions requestOptions = null,
             CancellationToken cancellationToken = default)
         {
+            // Currently, ReadItemAsync does not support the use of RequestOptions.IfMatchEtag property.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Container/ContainerInlineCore.cs:263]: I think Kiran meant on the public API of ReadItem, which is in Container (that is the public class). This comment here does not appear on the public docs

**Review by ealsur — CHANGES_REQUESTED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Container/ContainerInlineCore.cs (hunk 2 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/Resource/Container/ContainerInlineCore.cs]: Make sense. I have refactored the code to reflect the API doc update.

File: Microsoft.Azure.Cosmos/src/RequestOptions/RequestOptions.cs (hunk 1 above)
Comment (FabianMeiswinkel) [reply Microsoft.Azure.Cosmos/src/RequestOptions/RequestOptions.cs:21]: Patch I think also supports it?

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 3]
```diff
@@ -432,7 +432,8 @@ public abstract class Container
         /// Items contain meta data that can be obtained by mapping these meta data attributes to properties in <typeparamref name="T"/>.
         /// * "_ts": Gets the last modified time stamp associated with the item from the Azure Cosmos DB service.
         /// * "_etag": Gets the entity tag associated with the item from the Azure Cosmos DB service.
-        /// * "ttl": Gets the time to live in seconds of the item in the Azure Cosmos DB service.
+        /// * "ttl": Gets the time to live in seconds of the item in the Azure Cosmos DB service. Note that, this API does not support
+        /// the usage of <see cref="RequestOptions.IfMatchEtag"/> property at the moment.
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:436]:
```suggestion
        /// Note that, this API does not support the usage of <see cref="RequestOptions.IfMatchEtag"/> property at the moment.
```

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 4]
```diff
@@ -432,7 +432,8 @@ public abstract class Container
         /// Items contain meta data that can be obtained by mapping these meta data attributes to properties in <typeparamref name="T"/>.
         /// * "_ts": Gets the last modified time stamp associated with the item from the Azure Cosmos DB service.
         /// * "_etag": Gets the entity tag associated with the item from the Azure Cosmos DB service.
-        /// * "ttl": Gets the time to live in seconds of the item in the Azure Cosmos DB service.
+        /// * "ttl": Gets the time to live in seconds of the item in the Azure Cosmos DB service. Note that, this API does not support
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:435]:
```suggestion
        /// * "ttl": Gets the time to live in seconds of the item in the Azure Cosmos DB service.
```

**Review by ealsur — APPROVED** [review]
LGTM. Just left a nit regarding new line handling

File: Microsoft.Azure.Cosmos/src/RequestOptions/RequestOptions.cs (hunk 1 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/RequestOptions/RequestOptions.cs:21]: Not sure on that @FabianMeiswinkel. Do you want me to cover that as well? Cc: @ealsur , @kirankumarkolli.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/RequestOptions/RequestOptions.cs:21]: If Patch supports it, sure

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs (hunk 3 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: Updated.

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs (hunk 4 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs]: Updated.

**Review by ealsur — DISMISSED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs [hunk 5]
```diff
@@ -433,6 +433,7 @@ public abstract class Container
         /// * "_ts": Gets the last modified time stamp associated with the item from the Azure Cosmos DB service.
         /// * "_etag": Gets the entity tag associated with the item from the Azure Cosmos DB service.
         /// * "ttl": Gets the time to live in seconds of the item in the Azure Cosmos DB service.
+        /// Note that, this API does not support the usage of <see cref="RequestOptions.IfMatchEtag"/> property at the moment.
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:436]: Similar to above can we please move this to <remarks> section?

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs (hunk 5 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:436]: This is already in `<remarks>` what section would it go to?

File: Microsoft.Azure.Cosmos/src/RequestOptions/RequestOptions.cs (hunk 1 above)
Comment (kundadebdatta) [reply Microsoft.Azure.Cosmos/src/RequestOptions/RequestOptions.cs:21]: Added relevant documentation in `<remarks>` section.

File: Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs (hunk 5 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Resource/Container/Container.cs:436]: Apologies :-)

**Review by kirankumarkolli — APPROVED** [review]

**Review by ealsur — APPROVED** [review]

**Pull request merged**